src/main/scala/RockPaperScissors.scala

```diff
@@ -0,0 +1,188 @@
+    else if(playMode == 2)
+    runAIVSAI(totalGames)
+    else if(playMode == 3)
+    runLotsOfAI()
+
+
+
+
+
+  def takeInput(): String = readLine()
+
+  def generateAIMove(): Int = scala.util.Random.nextInt(totalGames * 10)
+
+  def createGroupSizeForWeights: Int = (totalGames * 10)/ 3
+
+
+  def commitAImove
+  (genratedNumber: Int , weightedSc: Int, weightedRock: Int ,weightedPaper: Int ,weightGroupsSizeO: Int ): String =
+    (genratedNumber) match
+  {
+    case (a)
+      if ( 0 < genratedNumber && genratedNumber <= (
+        weightGroupsSizeO - weightedRock - weightedSc + weightedPaper)) => "scissor"
+    case (a)
+      if ( weightGroupsSizeO < genratedNumber && genratedNumber <= (
+        (weightGroupsSizeO * 2) - weightedPaper - weightedSc + weightedRock)) => "paper"
+    case (a)
+      if ( (weightGroupsSizeO * 2)  < genratedNumber && genratedNumber <= (
+        (weightGroupsSizeO * 3)  - weightedPaper - weightedRock +weightedSc)) => "rock"
+    case  _ =>  commitAImove(generateAIMove(), weightedSc , weightedRock , weightedPaper ,weightGroupsSizeO)
+  }
+
+  def weightAdjustmentCheck(input: String): Unit = (input) match
+  {
+    case (input) if( input == "rock") => rockWeight += 1
+    case (input) if(input == "paper") =>  paperWeight += 1
+    case (input) if(input == "scissor") => scissorsWeight += 1
+  }
+
+
+  def determineWinner(moveAI: String, movePlayer: String): String = (moveAI) match
+  {
+    case(a) if(moveAI == movePlayer) => "draw against first A.I"
+    case(a) if(moveAI == "paper" && movePlayer == "rock") => winning -= 1; "lose against first A.I"
+    case(a) if(moveAI == "rock" && movePlayer == "scissor") => winning -= 1; "lose against first A.I"
+    case(a) if(moveAI == "scissor" && movePlayer == "paper") => winning -= 1;  "lose against first A.I"
+    case(a) if(moveAI == "rock" && movePlayer == "paper" ) => winning += 1; "win against first A.I"
+    case(a) if(moveAI == "scissor" && movePlayer =="rock" ) => winning += 1; "win against first A.I"
+    case(a) if(moveAI == "paper"&& movePlayer == "scissor") =>  winning += 1; "win against first A.I"
+
+  }
+
+  def determineOveralWinner(): Unit =
+  {
+     if(winning > 0) println("first player has won by " + winning)
+     else if (winning < 0) println("first player has lost by " + winning)
+     else println("neither player lost or won")
+  }
+
+  def runAIVSAI(totalGames: Int) =
+  {
+    var totalGamesInFunction = totalGames
+    while (totalGamesInFunction > 0)
+    {
+      var moveAI = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight ,weightGroupsSizeO )
+      var movePlayer = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight, weightGroupsSizeO )
+      weightAdjustmentCheck(movePlayer)
+      println(determineWinner(moveAI, movePlayer ))
+      println("genrated Ai move " + moveAI)
+      totalGamesInFunction -= 1
+    }
+    determineOveralWinner()
+  }
+
+  def runAIVSPlayer (totalGames: Int) =
```

Comment:
looks exactly the same as **runAIVSAI**, if so why do you need two methods that does the same job